code/_onclick/item_attack.dm

```diff
@@ -46,6 +46,8 @@
 /obj/item/proc/attack_obj(obj/O, mob/living/user)
 	if(flags & NOBLUDGEON)
 		return
+	if(user.a_intent == "help") //so you don't hit a machine with a tool while trying to disassemble it.
+		return

```

Comment:
Somewhat controversial change: no attack on objects if on help intent. (so you don't attack the machine you're trying to deconstruct with the right tool).